src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -77,6 +77,24 @@ namespace OpenMS
       bool operator()(const MSSpectrum& a, const MSSpectrum& b) const;
     };
 
+    /// Used to remember what subsets in a spectrum are sorted already to allow faster sorting of the spectrum
+    struct Chunk {
+      Size start;
+      Size end;
+      bool isSorted;
+      Chunk(Size a, Size b, bool c) : start(a), end(b), isSorted(c) {}
+    };
+
+    struct Chunks {
+      std::vector<Chunk> chunks_;
+      const MSSpectrum& spec_;
```

Comment:
`spec_` should be private?

Reply:
I guess yes, because otherwise one could change the associated spectrum during getSpectrum() (which wouldnt make sense, but still this can be prevented)